templates/utils.js

```diff
@@ -10,8 +10,13 @@ const replaceByClonedSource = options => {
 };
 
 const deepMergeWithoutKeysOrder = createDeepMerge({ mergeArray: replaceByClonedSource });
+const deepMergeArray = createDeepMerge();
 
 export const deepMerge = (...args) => {
+  if (Array.isArray(args[0])) {
```

Comment:
Also, I'd prefer to not do this inside current `deepMerge` function. Because
1) top level Array merge behavior is different from other Arrays merge behavior which is weird
2) what if I want to merge top level array with replace behavior?

So I think it's better to use another function for that case, for example `deepMergeConcat` https://github.com/scaffold-eth/create-eth/pull/247/files#r2052201977